File: mobile/android/base/ZoomConstraints.java
/* -*- Mode: Java; c-basic-offset: 4; tab-width: 20; indent-tabs-mode: nil; -*-
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.gecko;

import org.json.JSONException;
import org.json.JSONObject;

public final class ZoomConstraints {
    private final boolean mAllowZoom;
    private final boolean mAllowDoubleTapZoom;
    private final float mDefaultZoom;
    private final float mMinZoom;
    private final float mMaxZoom;

    public ZoomConstraints(boolean allowZoom) {
        mAllowZoom = allowZoom;
        mAllowDoubleTapZoom = allowZoom;
        mDefaultZoom = 0.0f;
        mMinZoom = 0.0f;
        mMaxZoom = 0.0f;
    }

    ZoomConstraints(JSONObject message) throws JSONException {
        mAllowZoom = message.getBoolean("allowZoom");
        mAllowDoubleTapZoom = message.getBoolean("allowDoubleTapZoom");
        mDefaultZoom = (float)message.getDouble("defaultZoom");
        mMinZoom = (float)message.getDouble("minZoom");
        mMaxZoom = (float)message.getDouble("maxZoom");
    }

    public final boolean getAllowZoom() {
        return mAllowZoom;
    }

    public final boolean getAllowDoubleTapZoom() {
        return mAllowDoubleTapZoom;
    }

    public final float getDefaultZoom() {
        return mDefaultZoom;
    }

    public final float getMinZoom() {
        return mMinZoom;
    }

    public final float getMaxZoom() {
        return mMaxZoom;
    }
}

File: gfx/layers/ZoomConstraints.cpp
/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
/* vim: set ts=8 sts=2 et sw=2 tw=80: */
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

#include "ZoomConstraints.h"

#include <ostream>
#include "nsPrintfCString.h"

namespace mozilla {
namespace layers {

std::ostream& operator<<(std::ostream& aStream, const ZoomConstraints& z) {
  return aStream << nsPrintfCString("{ z=%d, dt=%d, min=%f, max=%f }",
                                    z.mAllowZoom, z.mAllowDoubleTapZoom,
                                    z.mMinZoom.scale, z.mMaxZoom.scale)
                        .get();
}

}  // namespace layers
}  // namespace mozilla
